src/pipecat/utils/string.py

```diff
@@ -280,3 +280,55 @@ def append_part(part: TextPartForConcatenation):
     result = result.strip()
 
     return result
+
+
+def split_text_by_spaces(text: str) -> List[str]:
+    """Split text into individual words while preserving spacing.
+
+    Splits on spaces, preserving leading spaces on the first word and trailing
+    spaces on all words except the last. Words can be concatenated to reconstruct
+    the original text.
+
+    Args:
+        text: The text to split into words.
+
+    Returns:
+        A list of words with trailing spaces preserved (except the last word).
+        Leading spaces are preserved on the first word.
+
+    Example::
+
+        >>> split_text_by_spaces("Hello world!")
```

Comment:
Another thing we may want to handle: something like "Hello\nworld". 

Newline is not a space, but it maybe should be treated like one for the purpose of preparing the text for aggregation.

Also, maybe end-of-sentence characters should be considered delimiters, too, since that effects the `SimpleTextAggregator` aggregation. Like "Hello.World".

And actually, since we might be pattern matching on things like `<`, maybe those might need to be delimiters, too? Like "Hello<foo>world".

Seems like there are many many cases we might need to handle for this to really be robust enough to chunk text into sufficiently small pieces for any aggregator to handle it properly...unless...bear with me...what if we just fed text into the aggregator _character by character_ 🤯